src/components/medication-dispense-review.component.tsx

```diff
@@ -1,202 +1,277 @@
-import React, { useState } from "react";
+import React from "react";
 import { CommonConfigProps, MedicationDispense } from "../types";
-import DispenseCard from "../components/dispense-card.component";
-import {
-  Select,
-  SelectItem,
-  TextArea,
-  FormGroup,
-  RadioButton,
-  Toggle,
-  FormLabel,
-  DataTableSkeleton,
-  Column,
-  ComboBox,
-  Grid,
-  NumberInput,
-} from "@carbon/react";
-import styles from "./medication-dispense-review.scss";
+import MedicationCard from "./medication-card-component";
+import { TextArea, Column, ComboBox, Grid, NumberInput } from "@carbon/react";
 import { useLayoutType } from "@openmrs/esm-framework";
 import { useTranslation } from "react-i18next";
+import { getMedication } from "../utils";
+import styles from "./medication-dispense-review.scss";
 
 interface MedicationDispenseReviewProps {
   medicationDispense: MedicationDispense;
+  index: number;
+  updateMedicationDispense: Function;
   drugDosingUnits: Array<CommonConfigProps>;
+  drugDispensingUnits: Array<CommonConfigProps>;
   drugRoutes: Array<CommonConfigProps>;
   orderFrequencies: Array<CommonConfigProps>;
 }
 
 const MedicationDispenseReview: React.FC<MedicationDispenseReviewProps> = ({
   medicationDispense,
+  index,
+  updateMedicationDispense,
   drugDosingUnits,
+  drugDispensingUnits,
   drugRoutes,
   orderFrequencies,
 }) => {
   const { t } = useTranslation();
   const isTablet = useLayoutType() === "tablet";
-  const [dispenseRequest, setDispenseRequest] = useState(medicationDispense);
 
   return (
-    <div className={styles.reviewContainer}>
-      <DispenseCard medication={medicationDispense} />
+    <div className={styles.medicationDispenseReviewContainer}>
+      <MedicationCard medication={getMedication(medicationDispense)} />
 
-      <NumberInput
-        className="some-class"
-        id="tj-input"
-        invalidText="Number is not valid"
-        label="Quantity"
-        max={100}
-        min={0}
-        value={dispenseRequest.quantity.value}
-        onChange={(e) => {
-          setDispenseRequest({
-            ...dispenseRequest,
-            quantity: {
-              value: e.target.value,
-              unit: dispenseRequest.quantity.unit,
-              code: dispenseRequest.quantity.code,
-            },
-          });
-        }}
-        size="md"
-        step={1}
-        warnText="A high threshold may impact performance"
-      />
+      <Grid className={styles.noPadding}>
+        <Column lg={5}>
```

Comment:
I'm by no means a Carbon expert so not sure I got all this grid stuff right... I basically wrapped every row in it's own Grid, which was only way I could get each row properly left-aligned.

Reply:
I had refactored this to use flexbox so i'll be changing this after the rebase

Reply:
Awesome, thanks, please do that!